packages/graphql-js-operation-payload-generator/src/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { GraphQLList, getNullableType } from "graphql";
+import type { GraphQLOutputType } from "graphql";
+
+export function getArrayDepthFromType(type: GraphQLOutputType) {
+  let depth = 0;
+  let currentType = getNullableType(type);
+
+  while (currentType instanceof GraphQLList) {
+    depth += 1;
+    currentType = getNullableType(currentType.ofType);
+  }
+
+  return depth;
+}
+
+export function wrapInArray(value: unknown, depth: number) {
```

Comment:
can you give examples what's this used for? I read it as "burry element in nested arrays up to specified depth", but I can't really connect it to a real scenario. Does depth ever go above 2?

Reply:
My idea is to detect required depth for an array, without further limitation and hardcoding.
As it is open source, somebody maybe has a 3 dimensional array.

Reply:
I'm still can't connect it to a use case. Can you provide a schema example with such type?

Reply:
```
type Query {
        nestedList(id: ID!): [[CustomNode!]!]!
        deeplyNestedList(id: ID!): [[[CustomNode!]!]!]!
}
```